src/course-user/course-user.service.ts

```diff
@@ -1,4 +1,33 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CourseUserEntity } from '../entities/course-user.entity';
+import { CourseUserRepository } from './course-user.repository';
+import { CourseUserDto } from './dto/course-user.dto';
 
 @Injectable()
-export class CourseUserService {}
+export class CourseUserService {
+  constructor(
+    @InjectRepository(CourseUserRepository) private courseUserRepository: CourseUserRepository,
+  ) {}
+
+  async courseUserExists({ userId, courseId }: CourseUserDto): Promise<CourseUserEntity> {
```

Comment:
maybe for naming consistency, `courseUserExists` could be `getCourseUser` as it returns the record